cmd/outputs.go

```diff
@@ -0,0 +1,275 @@
+		RuntimeOptions: rtOpts,
+		ExecutionPlan:  executionPlan,
+	}
+	result := make([]output.Output, 0, len(outputFullArguments))
+
+	for _, outputFullArg := range outputFullArguments {
+		outputType, outputArg := parseOutputArgument(outputFullArg)
+		outputConstructor, ok := outputConstructors[outputType]
+		if !ok {
+			return nil, fmt.Errorf(
+				"invalid output type '%s', available types are: %s",
+				outputType, getPossibleIDList(outputConstructors),
+			)
+		}
+
+		params := baseParams
+		params.OutputType = outputType
+		params.ConfigArgument = outputArg
+		params.JSONConfig = conf.Collectors[outputType]
+
+		output, err := outputConstructor(params)
+		if err != nil {
+			return nil, fmt.Errorf("could not create the '%s' output: %w", outputType, err)
+		}
+		result = append(result, output)
+	}
+
+	return result, nil
+}
+
+func parseOutputArgument(s string) (t, arg string) {
+	parts := strings.SplitN(s, "=", 2)
+	switch len(parts) {
+	case 0:
+		return "", ""
+	case 1:
+		return parts[0], ""
+	default:
+		return parts[0], parts[1]
+	}
+}
+
+// TODO: remove this after we transition every collector to the output interface
+
+func newCollectorAdapter(params output.Params, collector lib.Collector) (output.Output, error) {
+	// Check if all required tags are present
+	missingRequiredTags := []string{}
+	requiredTags := collector.GetRequiredSystemTags()
+	for _, tag := range stats.SystemTagSetValues() {
+		if requiredTags.Has(tag) && !params.ScriptOptions.SystemTags.Has(tag) {
+			missingRequiredTags = append(missingRequiredTags, tag.String())
+		}
+	}
+	if len(missingRequiredTags) > 0 {
+		return nil, fmt.Errorf(
+			"the specified output '%s' needs the following system tags enabled: %s",
+			params.OutputType, strings.Join(missingRequiredTags, ", "),
+		)
+	}
+
+	return &collectorAdapter{outputType: params.OutputType, collector: collector}, nil
+}
+
+// collectorAdapter is a _temporary_ fix until we move all of the old
+// "collectors" to the new output interface
+type collectorAdapter struct {
+	collector    lib.Collector
+	outputType   string
+	runCtx       context.Context
+	runCtxCancel func()
+}
+
+func (ca *collectorAdapter) Description() string {
+	link := ca.collector.Link()
+	if link != "" {
+		return fmt.Sprintf("%s (%s)", ca.outputType, link)
+	}
+	return ca.outputType
+}
+
+func (ca *collectorAdapter) Start() error {
+	if err := ca.collector.Init(); err != nil {
+		return err
+	}
+	ca.runCtx, ca.runCtxCancel = context.WithCancel(context.Background())
+	go ca.collector.Run(ca.runCtx)
+	return nil
+}
+
+func (ca *collectorAdapter) AddMetricSamples(samples []stats.SampleContainer) {
+	ca.collector.Collect(samples)
+}
+
+func (ca *collectorAdapter) SetRunStatus(latestStatus lib.RunStatus) {
+	ca.collector.SetRunStatus(latestStatus)
+}
+
+// Stop implements the new output interface.
+func (ca *collectorAdapter) Stop() error {
+	ca.runCtxCancel()
+	time.Sleep(2 * time.Second) // shrug, better than nothing...
```

Comment:
:rofl:

Reply:
I mean you can wait for the `Run` from line 256 to end ... that will be better ;)

Reply:
True, I'll do that :+1:

Reply:
Fixed by https://github.com/loadimpact/k6/pull/1869/commits/e5afaa3393905be4e6d81ca53da838426d0a0df5